toolkit/tools/imagegen/configuration/configuration.go

```diff
@@ -92,13 +68,70 @@ type Config struct {
 	DefaultSystemConfig *SystemConfig // A system configuration with the "IsDefault" field set or the first system configuration if there is no explicit default.
 }
 
+// GetDiskPartByID returns the disk partition object with the desired ID, nil if no partition found
+func (c *Config) GetDiskPartByID(ID string) (disk *Partition) {
+	for i, d := range c.Disks {
+		for j, p := range d.Partitions {
+			if p.ID == ID {
+				return &c.Disks[i].Partitions[j]
+			}
+		}
+	}
+	return nil
+}
+
+// checkDeviceMapperFlags checks if Encryption and read-only roots have a the required 'dmroot' flag.
+// They need the root partition to have a specific flag so we can find the partition and handle it
+// before we mount it.
+func checkDeviceMapperFlags(config *Config) (err error) {
+	for _, sysConfig := range config.SystemConfigs {
+		var dmRoot *Partition
+		if sysConfig.ReadOnlyVerityRoot.Enable || sysConfig.Encryption.Enable {
+			rootPartSetting := sysConfig.GetRootPartitionSetting()
+			if rootPartSetting == nil {
+				return fmt.Errorf("can't find a root ('/') [PartitionSetting] to work with either [ReadOnlyVerityRoot] or [Encryption]")
+			}
+			rootDiskPart := config.GetDiskPartByID(rootPartSetting.ID)
+			if rootDiskPart == nil {
+				return fmt.Errorf("can't find a [Disk] [Partition] to match with [PartitionSetting] '%s'", rootPartSetting.ID)
+			}
+			if !rootDiskPart.HasFlag("dmroot") {
+				return fmt.Errorf("[Partition] '%s' must include 'dmroot' device mapper root flag in [Flags] for [SystemConfig] '%s's root partition since it uses [ReadOnlyVerityRoot] or [Encryption]", rootDiskPart.ID, sysConfig.Name)
```

Comment:
>return fmt.Errorf("[Partition] '%s' must includ [](start = 4, length = 47)

I like the detailed errors.

Reply:
Partly defense against someone using this stuff I don't know about. At least this way it will be obvious what they need to fix when they hit a build break.